src/webgpu/shader/execution/builtin/atan.spec.ts

```diff
@@ -0,0 +1,63 @@
+export const description = `
+Execution Tests for the 'atan' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, TypeF32 } from '../../../util/conversion.js';
+
+import { ulpThreshold, Case, Config, run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('float_builtin_functions,atan')
+  .uniqueId('b13828d6243d13dd')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#float-builtin-functions')
+  .desc(
+    `
+atan:
+T is f32 or vecN<f32> atan(e: T ) -> T Returns the arc tangent of e. Component-wise when T is a vector. (GLSLstd450Atan)
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    // TODO(https://github.com/gpuweb/cts/issues/792): Decide what the ground-truth is for these tests.
+    const truth_func = (x: number): number => {
+      return Math.atan(x);
+    };
+    // Well defined/border cases
+    const manual_inputs: Array<number> = [
+      Number.NEGATIVE_INFINITY, // -Inf -> -90°
```

Comment:
We seem to be mixing how we declare infinities. [David has been using `Infinity` and `-Infinity`](https://github.com/gpuweb/cts/blob/8d46b8304c5978490b941c41b521b8ae35b2d7db/src/webgpu/shader/execution/builtin/isinf.spec.ts#L36-L37), [I've been using `f32Bits(0x7f800000)` and `f32Bits(0xff800000)`](https://github.com/gpuweb/cts/pull/796/files), and you're using `Number.POSITIVE_INFINITY` and `Number.NEGATIVE_INFINITY`, and I just stumbled across Sarah's [`kBit.f32.infinity.positive` and `kBit.f32.infinity.negative`](https://github.com/gpuweb/cts/blob/8d46b8304c5978490b941c41b521b8ae35b2d7db/src/webgpu/shader/execution/builtin/builtin.ts#L549-L552)

I've no particular strong feeling which one is better, but we should probably be consistent. Sarah's constants predate all of ours, so maybe we should switch to using them. @dneto0 - thoughts?

David

Reply:
In a local patch I have converted the manual cases to use defined values, as requested in a related PR, i.e.
`{ input: f32(Number.NEGATIVE_INFINITY), expected: f32(-Math.PI / 2) },`

This passes, but if I change the input to `kBit.f32.infinity.negative`, the test starts failing because the return value is actually Pi/2 instead of -Pi/2.

I am unsure if this is an issue with the testing framework or TS/JS conversions.

Reply:
> This passes, but if I change the input to `kBit.f32.infinity.negative`, the test starts failing 

This is likely because these numbers are unsigned integer representations of the float values. You need to pass these to `f32Bits` not `f32`

Reply:
Yup that fixed it.